bofire/data_models/domain/features.py

```diff
@@ -622,6 +623,21 @@ def _validate_transform_specs(
                     raise ValueError(
                         f"Forbidden transform type for feature with key {key}",
                     )
+            checked_keys.add(key)
```

Comment:
From my understanding this whole part here is not necessary, as this is already checked above. Every feature has a method `valid_transform_types` and then it is checked if the assigned transform type is valid based on the registered ones in the feature.